File: main-api/app/models/comment.py
from datetime import datetime, timezone
from app import db


class Comment(db.Model):
    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    issue_id = db.Column(db.Integer, db.ForeignKey("issues.id", ondelete="CASCADE"), nullable=False)
    author_id = db.Column(db.Integer, db.ForeignKey("users.id", ondelete="SET NULL"), nullable=True)

    content = db.Column(db.Text, nullable=False)
    created_at = db.Column(db.DateTime, default=lambda: datetime.now(timezone.utc))

    def to_dict(self):
        return {
            "id": self.id,
            "issue_id": self.issue_id,
            "author_id": self.author_id,
            "author_name": self.author.name if self.author else None,
            "content": self.content,
            "created_at": self.created_at.strftime('%Y-%m-%dT%H:%M:%SZ') if self.created_at else None,
        }

File: ios/Axion/Axion/Models/Comment.swift
import Foundation

struct Comment: Codable, Identifiable {
    let id: Int
    let issueId: Int
    let authorId: Int
    let content: String
    let createdAt: String?
    let authorName: String?

    enum CodingKeys: String, CodingKey {
        case id, content
        case issueId = "issue_id"
        case authorId = "author_id"
        case createdAt = "created_at"
        case authorName = "author_name"
    }
}
